plugins/ResourceFetcherPlugin/main.js

```diff
@@ -0,0 +1,103 @@
+
+// This is a sample implementation of a resource fetcher plugin.  It is intended to show how a resource fetcher plugin might be written.
+// This plugin uses an ajax call to call to a service to get the contents of a resource.
+//
+// It's based on a plugin used at Bibliotheca to get the contents of an epub from a server.  I've edited this file to remove
+// some functionality specific to Bibliotheca.  The result may or may not be suitiable for your needs.
+
+define(['readium_js_plugins', 'readium_shared_js/globals', '../../../js/epub-fetch/discover_content_type' ],
```

Comment:
`../../../js/epub-fetch/discover_content_type`
https://github.com/readium/readium-js/blob/develop/js/epub-fetch/discover_content_type.js

This reference is problematic, because code in `readium-shared-js` cannot make assumptions about its integration context. This dependency injection assumes that `readium-js` is the parent repository (that `readium-shared-js` is a submodule). ReadiumSDK-based native apps also make use of `readium-shared-js`, but have absolutely zero-awareness of `readium-js`.

Now, this issue is mitigated by the fact that in practice, the `ResourceFetcherPlugin` would only ever be used in the context of an app that relies on `readium-js`. But generally-speaking, there is definitely a problem in `readium-shared-js` plugins having dependencies from external RequireJS namespaces (`../../../js/epub-fetch/discover_content_type` is of course equivalent to `readium_js/epub-fetch/discover_content_type`, but this syntax would only work when building from `readium-js` or `readium-js-viewer`, not from within `readium-shared-js`).